pypeit/core/coadd.py

```diff
@@ -1390,179 +1557,135 @@ def ech_coadd(files,objids=None,extract='OPT',flux=True,giantcoadd=False,ordersc
-                      'v_pix': v_pix}
-        kwargs.update(ech_kwargs)
-        # Coadding
-        spec1d = coadd_spectra(spectra, wave_grid_method=wave_grid_method, niter=niter,
-                                          scale_method=scale_method, do_offset=do_offset, sigrej_final=sigrej_final,
-                                          do_var_corr=do_var_corr, qafile=qafile, outfile=outfile,
-                                          do_cr=do_cr, debug=debug,**kwargs)
-    else:
-        msgs.info('Coadding individual orders first and then merge order')
-        spectra_list = []
-        # Keywords for Table
-        rsp_kwargs = {}
-        rsp_kwargs['wave_tag'] = '{:s}_WAVE'.format(extract)
-        rsp_kwargs['flux_tag'] = '{:s}_FLAM'.format(extract)
-        rsp_kwargs['sig_tag'] = '{:s}_FLAM_SIG'.format(extract)
-        #wave_grid = np.zeros((2,norder))
-        for iord in range(norder):
-            spectra = load.ech_load_spec(files, objid=objids, order=iord, extract=extract, flux=flux)
-            ech_kwargs = {'echelle': False, 'wave_grid_min': spectra.wvmin.value, 'wave_grid_max': spectra.wvmax.value, 'v_pix': v_pix}
-            #wave_grid[0,iord] = spectra.wvmin.value
-            #wave_grid[1,iord] = spectra.wvmax.value
+    if nfile>1:
+        msgs.info('Coadding {:} spectra.'.format(nfile))
+        fname = files[0]
+        ext_final = fits.getheader(fname, -1)
+        norder = ext_final['ECHORDER'] + 1
+        msgs.info('spectrum {:s} has {:d} orders'.format(fname, norder))
+        if norder <= 1:
+            msgs.error('The number of orders have to be greater than one for echelle. Longslit data?')
+
+        if giantcoadd:
+            msgs.info('Coadding all orders and exposures at once')
+            spectra = load.ech_load_spec(files, objid=objids, order=None, extract=extract, flux=flux)
+            wave_grid = np.zeros((2, spectra.nspec))
+            for i in range(spectra.nspec):
+                wave_grid[0, i] = spectra[i].wvmin.value
+                wave_grid[1, i] = spectra[i].wvmax.value
+            ech_kwargs = {'echelle': True, 'wave_grid_min': np.min(wave_grid), 'wave_grid_max': np.max(wave_grid),
+                          'v_pix': v_pix}
             kwargs.update(ech_kwargs)
-            # Coadding the individual orders
-
-            if qafile is not None:
-                if len(qafile.split('.')) == 1:
-                    msgs.info("No fomat given for the qafile, save to PDF format.")
-                    qafile_iord = qafile + '.pdf'
+            # Coadding
+            spec1d_final = coadd_spectra(spectra, wave_grid_method=wave_grid_method, niter=niter,
+                                   scale_method=scale_method, do_offset=do_offset, sigrej_final=sigrej_final,
+                                   do_var_corr=do_var_corr, qafile=qafile, outfile=outfile,
+                                   do_cr=do_cr, debug=debug, **kwargs)
+            return spec1d_final
+        else:
```

Comment:
There's no need for this `else`.  The function will return if it entered the `if` block.